TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer

```diff
@@ -0,0 +1,317 @@
+package org.firstinspires.ftc.teamcode.pedroPathing.localization.localizers;
+
+import android.util.Size;
+
+import com.acmerobotics.roadrunner.Vector2d;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
+import org.firstinspires.ftc.robotcore.external.matrices.VectorF;
+import org.firstinspires.ftc.robotcore.external.navigation.DistanceUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.Quaternion;
+import org.firstinspires.ftc.teamcode.pedroPathing.localization.Localizer;
+import org.firstinspires.ftc.teamcode.pedroPathing.localization.Pose;
+import org.firstinspires.ftc.teamcode.pedroPathing.pathGeneration.MathFunctions;
+import org.firstinspires.ftc.teamcode.pedroPathing.pathGeneration.Vector;
+import org.firstinspires.ftc.vision.VisionPortal;
+import org.firstinspires.ftc.vision.apriltag.AprilTagDetection;
+import org.firstinspires.ftc.vision.apriltag.AprilTagLibrary;
+import org.firstinspires.ftc.vision.apriltag.AprilTagProcessor;
+
+import java.util.ArrayList;
+
+/**
+ * This is the April Tag Localizer class. This class extends the Localizer superclass and is a
+ * localizer that uses a camera to read april tags to always know where your robot is on the field.
+ * This class also uses a secondary localizer to gauge heading and for when your camera can't see
+ * any april tags.
+ *
+ * The diagram below, which is taken from Road Runner, shows a typical set up.
+ *
+ * The view is from the bottom of the robot looking upwards.
+ *
+ * left on robot is y pos
+ *
+ * front on robot is x pos
+ *
+ *    /--------------\
+ *    |     ____     |
+ *    |     ----     |
+ *    | ||        || |
+ *    | ||        || |   left (y pos)
+ *    |              |
+ *    |              |
+ *    \--------------/
+ *      front (x pos)
+ *
+ * @author Anyi Lin - 10158 Scott's Bots
+ * @version 1.0, 7/20/2024
+ */
+
+/**
+ * Copyright (c) 2024, Austin Bacher @hypocritical1
+ * All rights reserved.
```

Comment:
Change this to the name of the license you are releasing under. Looks like you're using BSD-3-Clause?